src/derive_utils.rs

```diff
@@ -207,3 +207,24 @@ where
         <R as std::convert::TryFrom<&'a PyCell<T>>>::try_from(cell)
     }
 }
+
+/// Trait to abstract over the arguments of Python function wrappers.
+#[doc(hidden)]
+pub trait WrapPyFunctionArguments<'a, 'b> {
```

Comment:
Is there a reason to have two lifetime arguments in this trait? If we set them to the same, then Rust can just shorten the longer lifetime to match, which seems good enough to me. 

---

Also, looking again at this trait, I noticed a different pattern to achieve the same thing:

```
pub enum WrapPyFunctionArguments<'a> {
    Python(Python<'a>),
    Module(&'a PyModule)
}
```

And can then just implement `Into<WrapPyFunctionArguments>` for each of `Python` and `&PyModule`.

I think that _maybe_ the enum is less confusing to understand in the long run, I'm also happy to leave it as-is (as long as we take the second lifetime argument out). Just an idea in cause you find you prefer one design over the other.

Reply:
> Is there a reason to have two lifetime arguments in this trait? If we set them to the same, then Rust can just shorten the 
> longer lifetime to match, which seems good enough to me.

That's weird, I was trying that for almost an hour, now I went back to verify that it in fact doesn't work...and...it compiles!

I'll check how nice the enum variant turns out